addons/web/static/src/js/view_form.js

```diff
@@ -6127,16 +6126,15 @@ instance.web.form.FieldMonetary = instance.web.form.FieldFloat.extend({
             this.set({"currency_info": null});
             return;
         }
-        return this.ci_dm.add(self.alive(new instance.web.Model("res.currency").query(["symbol", "position"])
-            .filter([["id", "=", self.get("currency")]]).first())).then(function(res) {
-            self.set({"currency_info": res});
-        });
+        return self.set({"currency_info": instance.web.get_currency(self.get("currency"))})
     },
     parse_value: function(val, def) {
-        return instance.web.parse_value(val, {type: "float", digits: (this.node.attrs || {}).digits || this.field.digits}, def);
+        var digits_precision = this.node.attrs.digits || this.field.digits || (this.get('currency_info') && this.get('currency_info').digits)
```

Comment:
Missing semicolon. Also extraction of digits_precision could be a separated method for clarity.